src/garage/tf/optimizers/conjugate_gradient_optimizer.py

```diff
@@ -66,8 +70,15 @@ def eval(x):
 
         return eval
 
+    def __getstate__(self):
+        """Object.__getstate__."""
+        new_dict = self.__dict__.copy()
+        del new_dict['opt_fun']
+        return new_dict
```

Comment:
Just curious, will `opt_fun` be reconstructed when unpickled?

Reply:
No. Unpickling a pickled optimizer object will not reconstruct `_opt_fun`. It needs to be created by calling the `update_opt` function with the parameters. But let's say a primitive uses the optimizer, pickling and unpickling a primitive calls the `update_opt` with the right parameters, and thus we have an equivalent object after de-serialization.  This behavior is consistent with the master branch (with Serializable).

Reply:
Should we put some guards in the optimizers to make sure they are properly reconstructed? We are assuming user will always call `update_opt` afterwards. Or at least, we should have an integration test to make sure `update_opt` will be called afterwards.

Reply:
Yes, currently we always assume the user calls `update_opt` before using other functions of the optimizer. I think it might be a good idea to raise some exception when the user does not set the parameters first.